src/core/resources/_selectors/nfts.ts

```diff
@@ -16,76 +19,64 @@ export type NFTCollectionSectionData = {
 export const selectNfts = (data?: NFTInfiniteData) =>
   data?.pages?.map((page) => page.nfts).flat();
 
-export const sortSectionsAlphabetically = (
-  sections: NFTCollectionSectionData[],
+export const selectNftCollections = (
+  data?: NFTCollectionInfiniteData,
+  hiddenNfts?: Record<string, boolean>,
 ) => {
-  return sections.sort((a, b) => {
-    const aName = a.collection.name.toLowerCase();
-    const bName = b.collection.name.toLowerCase();
-    if (aName < bName) {
-      return -1;
-    }
-    if (aName > bName) {
-      return 1;
-    }
-    return 0;
-  });
-};
-
-export const sortSectionsByRecent = (sections: NFTCollectionSectionData[]) => {
-  return sections.sort((a, b) => {
-    const earliestDate = new Date(-8640000000000000);
-    const aCollectionAcquisition = a.lastCollectionAcquisition;
-    const bCollectionAcquisition = b.lastCollectionAcquisition;
-    const dateA = aCollectionAcquisition
-      ? new Date(aCollectionAcquisition)
-      : earliestDate;
-    const dateB = bCollectionAcquisition
-      ? new Date(bCollectionAcquisition)
-      : earliestDate;
-    return dateB.getTime() - dateA.getTime();
-  });
-};
-
-export const selectNftCollections = (data?: NFTInfiniteData) => {
-  const nfts = selectNfts(data);
-  const collections =
-    nfts?.reduce(
-      (collections, nft) => {
-        const currentCollectionId = nft.collection.collection_id;
-        if (currentCollectionId) {
-          const existingCollection = collections[currentCollectionId];
-          if (existingCollection) {
-            existingCollection.assets.push(nft);
-          } else {
-            collections[currentCollectionId] = {
-              assets: [nft],
-              collection: nft.collection,
-              lastCollectionAcquisition: nft.last_collection_acquisition,
-            };
-          }
-        }
-        return collections;
+  let idsOfCollectionsWithHiddenNfts: string[] = [];
+  return data?.pages
+    ?.map((page) => page.collections)
+    .flat()
+    .filter((c) => {
+      const collectionNftIds = c.nft_ids
+        .map((s) => s.split('.'))
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        .filter(([_, id, num]) => !hiddenNfts?.[`${id}_${num}`]);
+      if (c.nft_ids.length !== collectionNftIds.length) {
+        idsOfCollectionsWithHiddenNfts = [
+          ...idsOfCollectionsWithHiddenNfts,
+          c.collection_id,
+        ];
+      }
+      return Boolean(
+        c.collection_details.name &&
+          c.collection_details.image_url &&
+          collectionNftIds?.length,
+      );
+    })
+    .concat({
+      collection_id: idsOfCollectionsWithHiddenNfts.join(','),
+      distinct_nfts_owned: Object.values(hiddenNfts || {}).filter((v) => v)
+        .length,
+      distinct_nfts_owned_string: '0',
+      nft_ids: [],
+      collection_details: {
```

Comment:
do we need this object if everything is empty?

Reply:
Some of the values are necessary to display a section, the hidden one in this case, the remaining empty values were placed there to avoid making this array multiple types.